### ISSUE TITLE
false positive: indexing into iterator

### ISSUE THREAD

**Issue by mkborregaard** [tonyhffong/Lint.jl#196]
This
```julia
a = Dict(:b => 2)
keys(a)[1]
```
gives this spurious error 
<img width="593" alt="skaermbillede 2017-02-21 kl 15 16 12" src="https://cloud.githubusercontent.com/assets/8429802/23168515/b6a00516-f848-11e6-979a-c06c0c7b19e3.png">
(btw it would be nice to be able to copy the linter errors)

**Comment by TotalVerb** [comment]
At least in standard Base julia, this error is not spurious:

```julia
julia> a = Dict(:b => 2)
Dict{Symbol,Int64} with 1 entry:
  :b => 2

julia> keys(a)[1]
ERROR: MethodError: no method matching getindex(::Base.KeyIterator{Dict{Symbol,Int64}}, ::Int64)
```

Is there a package that adds this definition?

**Comment by mkborregaard** [comment]
Holy sh*t that's right. I wasn't thinking.
